MoltenVK/MoltenVK/Utility/MVKFoundation.h

```diff
@@ -339,11 +339,13 @@ static inline bool mvkVkComponentMappingsMatch(VkComponentMapping cm1, VkCompone
 #define mvkPrintSizeOf(type)    printf("Size of " #type " is %lu.\n", sizeof(type))
 
 
-#pragma mark -
-#pragma mark Template functions
-
 #pragma mark Math
 
+/** Rounds the value to nearest integer using half-to-even rounding. */
+static inline double mvkRoundHalfToEven(const double val) {
```

Comment:
Oh, I forgot to mention: this could probably be a `constexpr` function.

Reply:
Good idea. I've added a separate commit to this PR to fix this, and to also similarly change a bunch of other inlines that could possibly be resolved at compile-time.

Reply:
Huh. It looks like Xcode 11 and 12 are too old to recognize `mvkRoundHalfToEven()` as `constexpr`.

Reply:
Also, if the functions aren't `static` and don't wind up inlined, won't they appear in the public symbol table? I guess it doesn't matter much, since the ODR requires that every instance of such a function have the same definition, and they all have prefixes which makes a name collision unlikely anyway.

Reply:
> Huh. It looks like Xcode 11 and 12 are too old to recognize `mvkRoundHalfToEven()` as `constexpr`.

Almost certainly because `std::remainder()` isn't `constexpr` in these versions.

Reply:
> Also, if the functions aren't static and don't wind up inlined, won't they appear in the public symbol table?

Ah. Thanks for catching that.

Are you thinking we should use `static constexpr` instead of `constexpr`, to ensure they are always inlined? If so...I guess there's no reason not to. 

In current practice it shouldn't make a difference. The `MVKFoundation.h` header is not meant for external consumption, and none of its functions are marked with `MVK_PUBLIC_SYMBOL`, so they don't survive the `GCC_SYMBOLS_PRIVATE_EXTERN=1` we use.

However, I guess the code itself shouldn't depend on how we build, and so we should be deliberate about marking things static when they should be. I'll add the `static` part back.

Reply:
> Almost certainly because `std::remainder()` isn't `constexpr` in these versions.

Hmmm...`std::remainder()` isn't `constexpr` [until](https://en.cppreference.com/w/cpp/numeric/math/remainder) C++23. Strange that Xcode 14 doesn't complain. I've reverted to `static inline`.